custom_components/tahoma/config_flow.py

```diff
@@ -130,16 +126,17 @@ async def async_step_init(self, user_input=None):
     async def async_step_update_interval(self, user_input=None):
         """Manage the options regarding interval updates."""
         if user_input is not None:
-            self.options[CONF_UPDATE_INTERVAL] = user_input[CONF_UPDATE_INTERVAL]
-            return self.async_create_entry(title="", data=self.options)
+            return self.async_create_entry(title="", data=user_input)
 
         return self.async_show_form(
             step_id="update_interval",
             data_schema=vol.Schema(
                 {
                     vol.Required(
                         CONF_UPDATE_INTERVAL,
-                        default=self.options.get(CONF_UPDATE_INTERVAL),
+                        default=self.config_entry.options.get(
+                            CONF_UPDATE_INTERVAL, DEFAULT_UPDATE_INTERVAL
+                        ),
```

Comment:
What’s the difference?

Reply:
Fallback to `DEFAULT_UPDATE_INTERVAL`, instead of setting that in the constructor.